pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -336,6 +368,100 @@ func (o *EditOptions) Run() error {
 	}
 }
 
+func (o *EditOptions) applyEditPatch(oringalInfos, infos []*resource.Info) error {
+	for ix := range infos {
+		updatedInfo := infos[ix]
+		originalJS, editedJS, err := SerializationConvert(o.Encoder, oringalInfos[ix].Object, updatedInfo.Object)
+		if err != nil {
+			return err
+		}
+
+		if reflect.DeepEqual(originalJS, editedJS) {
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "skipped")
+		} else {
+			o.annotationPatch(updatedInfo)
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "edited")
+		}
+	}
+	return nil
+}
+
+func (o *EditOptions) annotationPatch(update *resource.Info) error {
+	patch, _, err := GetApplyPatch(update, o.Encoder, false)
+	if err != nil {
+		return err
+	}
+	mapping := update.ResourceMapping()
+	client, err := o.f.UnstructuredClientForMapping(mapping)
+	if err != nil {
+		return err
+	}
+	helper := resource.NewHelper(client, mapping)
+	_, err = helper.Patch(o.CmdNamespace, update.Name, types.StrategicMergePatchType, patch)
+	if err != nil {
+		return preservedFile(err, o.file, o.ErrOut)
+	}
+	return nil
+}
+
+func GetApplyPatch(info *resource.Info, codec runtime.Encoder, isLocal bool) ([]byte, []byte, error) {
+	wrapper := func(data []byte) ([]byte, error) {
+		if len(data) == 0 {
+			return []byte("{}"), nil
+		}
+		objMap := map[string]map[string]map[string]string{}
+		metadataMap := map[string]map[string]string{}
+		annotationsMap := map[string]string{}
+		annotationsMap[annotations.LastAppliedConfigAnnotation] = string(data)
+		metadataMap["annotations"] = annotationsMap
+		objMap["metadata"] = metadataMap
+		jsonString, err := json.Marshal(objMap)
+		return jsonString, err
+	}
+
+	if info == nil {
+		return []byte("{}"), []byte(""), nil
+	}
+	var obj runtime.Object
+	if isLocal {
+		obj = info.VersionedObject
+	} else {
+		obj = info.Object
+	}
+	data, err := runtime.Encode(codec, obj)
+	if err != nil {
+		return nil, data, err
+	}
+	js, err := wrapper(data)
+	return js, data, err
+}
+
+func SerializationConvert(codec runtime.Encoder, original, updated runtime.Object) ([]byte, []byte, error) {
```

Comment:
probably not worth a helper function for this... I'd just call encodeToJson twice